### PR TITLE
Improve application list parsing

### PR DESCRIPTION
**Pull request by lathiat**
Improve the application list parsing using python+yaml in preparation
for further enhancements to the unit placement handling.

### CODE REVIEW

File: common/generate_bundle_base [hunk 1]
```diff
@@ -174,15 +186,12 @@ if ${MASTER_OPTS[HYPERCONVERGED_DEPLOYMENT]}; then
     # add default binding to all applications
     bindings="\    bindings:\n      '': ${MASTER_OPTS[DEFAULT_BINDING]}"
     find $bundles_dir -name \*.yaml| xargs -l sed -i -r "/^\s+charm:.+/a$bindings"
+else
+   echo test
```
Comment (nicolasbock) [inline common/generate_bundle_base:190]: Why do we need this debug statement?

File: common/generate_bundle_base [hunk 2]
```diff
@@ -150,7 +162,7 @@ if ${MASTER_OPTS[HYPERCONVERGED_DEPLOYMENT]}; then
     # detect all apps used and generate placement info by doing:
     #  * search for all app names in generated bundles/overlays
     #  * search unit_placement template with same name
-    for app in `egrep -hr "^\s+[0-9a-z\-]+:(\s*#*.*)$" $bundles_dir --exclude-dir=p| sed -rn 's/\s+([[:alnum:]\-]+):.*/\1/p'| sort -u`; do
+    for app in ${application_list[@]}; do
```
Comment (nicolasbock) [inline common/generate_bundle_base:165]: Using your Python script and dropping that horrible regex is very nice :smile:

**Comment by nicolasbock** [comment]
@lathiat, is this ready for merging or are you still working on this change? I am not clear on this based on the commit message.

File: common/generate_bundle_base (hunk 1 above)
Comment (lathiat) [reply common/generate_bundle_base]: Dropped it :)

**Comment by lathiat** [comment]
Ready for review and merge

**Pull request merged**